packages/autocomplete-core/src/onInput.ts

```diff
@@ -115,11 +115,23 @@ export function onInput<TItem extends BaseItem>({
       })
   )
     .then((collections) => {
+      // Parameters passed to `onInput` could be stale when the following code
+      // executes, because `onInput` calls may not resolve in order.
+      // If it becomes a problem we'll need to save the last passed parameters.
+      // See: https://codesandbox.io/s/agitated-cookies-y290z
+
       setStatus('idle');
+
+      if (store.shouldSkipPendingUpdate) {
```

Comment:
As discussed in DM this branch doesn't cover this scenario:

```
+--------------------+
| q1 +---------->    |
| q2 +-------------> |
| q3 +-----> SUBMIT  |
+--------------------+
```

`q1` will get ignored, but `q2` will kick in because we only skip the next pending update, whereas it should be cancelled because `q3` was already submitted.

Reply:
Addressed in ac2b33f (+ test refactoring for clarity).